src/Examples/Flowsheets/ngcc/soec_flowsheet_costing.py

```diff
@@ -0,0 +1,291 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+__author__ = "Alex Noring"
+
+import pyomo.environ as pyo
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+from idaes.models_extra.power_generation.costing.power_plant_costing import (
+    get_PP_costing,
+    get_total_TPC,
+    costing_initialization,
+    get_fixed_OM_costs,
+    get_variable_OM_costs,
+    initialize_fixed_OM_costs,
+    initialize_variable_OM_costs,
+)
+
+
+def add_total_plant_cost(b, project_contingency, process_contingency):
+
+    # b.costing.total_plant_cost = pyo.Var(bounds=(0, 1e4))
+    try:
+        b.parent_block().generic_costing_units.append(b)
+    except AttributeError:
+        b.parent_block().generic_costing_units = []
+        b.parent_block().generic_costing_units.append(b)
+
+    @b.costing.Expression()
+    def total_plant_cost(c):
+        return c.purchase_cost * project_contingency * process_contingency / 1e6
+
+
+def get_solo_soec_capital_costing(m):
+    m.get_costing(year="2018")
```

Comment:
This is probably a topic for a separate issue and PR, but this appears to be using the old-style costing approach which is going to be removed in November.